src/runtime/sql-refinement-generator.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * @license
+ * Copyright (c) 2020 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {Op} from './manifest-ast-nodes.js';
+import {Dictionary} from './hot.js';
+import {Schema} from './schema.js';
+import {RefinementExpressionVisitor, BinaryExpression, UnaryExpression, FieldNamePrimitive, QueryArgumentPrimitive, BuiltIn, NumberPrimitive, BooleanPrimitive, TextPrimitive, Primitive} from './refiner.js';
+
+const sqlOperator: Dictionary<string> = {
+  [Op.AND]: 'AND',
+  [Op.OR]: 'OR',
+  [Op.LT]: '<',
+  [Op.GT]: '>',
+  [Op.LTE]: '<=',
+  [Op.GTE]: '>=',
+  [Op.ADD]: '+',
+  [Op.SUB]: '-',
+  [Op.MUL]: '*',
+  [Op.DIV]: '/',
+  [Op.NOT]: 'NOT',
+  [Op.NEG]: '-',
+  [Op.EQ]: '=',
+  [Op.NEQ]: '<>',
+};
+
+class SqlRefinementGenerator extends RefinementExpressionVisitor<string> {
+
+  visitBinaryExpression(expr: BinaryExpression): string {
+    return `(${this.visit(expr.leftExpr)} ${sqlOperator[expr.operator.op]} ${this.visit(expr.rightExpr)})`;
+  }
+  visitUnaryExpression(expr: UnaryExpression): string {
+    return `(${sqlOperator[expr.operator.op]} ${this.visit(expr.expr)})`;
+  }
+  visitFieldNamePrimitive(expr: FieldNamePrimitive): string {
+    const fixed = (expr.value.toString());
+    if (expr.evalType === Primitive.BOOLEAN) {
+      return `(${fixed} = 1)`;
+    }
+    return fixed;
+  }
+  visitQueryArgumentPrimitive(expr: QueryArgumentPrimitive): string {
+    return expr.value.toString();
+  }
+  visitBuiltIn(expr: BuiltIn): string {
+    if (expr.value === 'now()') {
+      return expr.value;
+    }
+    // TODO: Implement SQL getter for 'creationTimeStamp'
+    throw new Error(`Unhandled BuiltInNode '${expr.value}' in toSQLExpression`);
+  }
+  visitNumberPrimitive(expr: NumberPrimitive): string {
+    return expr.value.toString();
+  }
+  visitBooleanPrimitive(_: BooleanPrimitive): string {
```

Comment:
You mean you can't write foo.bar == true?

Reply:
maybe it would be normalized to `foo.bar` in SQL? and `foo.bar == false` would become `NOT foo.bar` ?

Reply:
The refinement system would rewrite it automatically. We could add support for it, but it just seems like another edge case to update.